src/lib/requests.ts

```diff
@@ -9,7 +9,8 @@ const logger = new Logger("requests")
 
 class DefaultMap extends Map {
     constructor(private defaultFactory, ...args) {
-        super(...args)
+        //super(...args)
+        super()
```

Comment:
This is the line in question.

Reply:
Just realized that Map() takes a [single argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), so
```suggestion
constructor(private defaultFactory, args) {
        super(args)
```
Should preserve original behavior and be more correct.

Reply:
Cool. Can't use that suggestion as it only deletes one line. Will do it the old fashioned way.

Reply:
I have no idea how to remove multiple lines with a single suggestion, do you know if it's possible?

Reply:
It isn't possible. Make multiple suggestions. As I said, I'll do it the old fashioned way, or you can, I think you can push to this branch.

Reply:
```
Successfully installed Tridactyl native messenger!
Run ':native' in Firefox to check.
ℹ ｢atl｣: Using typescript@3.3.3333 from typescript
ℹ ｢atl｣: Using tsconfig.json from /home/olie/projects/tridactyl/tsconfig.json
ℹ ｢atl｣: Checking started in a separate process...
✖ ｢atl｣: Checking finished with 1 errors

ERROR in [at-loader] ./src/lib/requests.ts:12:9
    TS2554: Expected 0 arguments, but got 1.

```

No cigar :(

Reply:
That's weird. I guess the solution is to go back to prototype-based definitions and do something like this:
```
let DefaultMap = function (defaultFactory, ...args) {
  Map.prototype.constructor.call(this, args)
  this.defaultFactory = defaultFactory;
}
DefaultMap.prototype = Object.create(Map.prototype)
DefaultMap.prototype.constructor = DefaultMap
DefaultMap.prototype.get = function (key) {
  if (!this.has(key)) {
    this.set(key, this.defaultFactory(key))
  }
  return Map.prototype.get.call(this, key)
}
```